builtin/logical/transit/lock_manager.go

```diff
@@ -0,0 +1,337 @@
+package transit
+
+import (
+	"encoding/json"
+	"fmt"
+	"sync"
+
+	"github.com/hashicorp/vault/logical"
+)
+
+const (
+	shared    = false
+	exclusive = true
+)
+
+type lockManager struct {
+	// A lock for each named key
+	locks map[string]*sync.RWMutex
+
+	// A mutex for the map itself
+	lockMutex sync.RWMutex
+
+	// If caching is enabled, the map of name to in-memory policy cache
+	cache map[string]*Policy
+
+	// Used for global locking, and as the cache map mutex
+	globalMutex sync.RWMutex
```

Comment:
Given the duration of the `lockMutex` and there is no interleaving between the lock domain for `lockMutex` and `globalMutex`, why not collapse these into the same lock?

If the `lockManager` has two assets it is protecting, `locks` and `cache`, can this be renamed to `cacheRWLock` just like the above `locksRWLock`?